### DISCUSSION TITLE
How to modify the style of DataTemplate in ControlTheme

### DISCUSSION THREAD

**Discussion by qqJoker** [AvaloniaUI/Avalonia#16518] [Q&A]
I have a ListBox on the interface, in which I have defined a `DataTemplate` with `x:key="MainMenuItemTemplate"` in the resource file. How can I use the pseudo class in `ControlTheme ` and modify the background and foreground colors of the Border control in the DataTemplate when the mouse moves into the ListBoxItem?

![1722238242214_981B83AB-6DC2-466a-8F14-7F0A8AC2BB82](https://github.com/user-attachments/assets/a9c73a86-11cc-4325-a13f-c75140eda57c)

**Comment by maxkatz6** [comment]
DataTemplate is independent from the ControlTheme.
But you always can add a Style inside of DataTemplate (in your example, inside of MainMenuBorder), and write a full selector like `ListBoxItem:pointerover Border#MainMenuBorder`

**Reply by qqJoker** [reply]
I tried adding a Style selector to Border, but it didn't seem to have any effect.
![image](https://github.com/user-attachments/assets/070b5b40-308d-45a4-a08a-007d12ee65c2)

**Reply by maxkatz6** [reply]
You set foreground with a local value on the TextBlock directly. It will have higher priority than anything else.
You need to add second style without pseudoclass for a default value.
